eos-updater-prepare-volume/eos-updater-prepare-volume

```diff
@@ -173,6 +174,17 @@ class VolumePreparer:
 
         return runtimes
 
+    def _add_autoinstall_flatpaks(self, flatpak_collection_refs):
+        all_actions = self.__run(['/usr/libexec/eos-updater-flatpak-installer', '--mode=print'])
+        for line in all_actions.split('\n'):
+            pattern = '^ *- (install|update) \(collection-id: (?P<cid>.*?)\|.*?\):(?P<ref>.*?)$'
+            match = re.search(pattern, line)
+            if match is not None:
+                ref_tuple = (match.group('cid'), match.group('ref'))
+                if self.debug:
+                    print('Adding a Flatpak autoinstall app: ' + str(ref_tuple))
+                flatpak_collection_refs.append(ref_tuple)
```

Comment:
Oh, I see how this is being used now.

Instead of parsing the output of `--print` (which was adapted from the existing debug messages), perhaps it might be better to have `eos-updater-flatpak-installer` dump JSON and use that instead?

Reply:
I was also thinking that perhaps `--print` might not belong in `eos-updater-flatpak-installer`. The good news is that `eufi_determine_flatpak_ref_actions_to_check` and friends are already in a separate library, so you could create a new binary and link directly to that. Not sure if we want yet another binary though. @pwithnall any thoughts on that?

Reply:
FWIW, I agree with @pwithnall , though I don't know if libeos-updater-flatpak-installer is introspected yet. Shouldn't be too bad to introspect it, though it does use GPtrArray quite heavily and not sure if the element types would need to be boxed.